src/content/posts/2023-03-09-creating-a-pagination-component-with-astro.md

```diff
@@ -0,0 +1,288 @@
+---
+title: "Creating A Pagination Component With Astro"
+slug: creating-a-pagination-component-with-astro
+date: 2023-03-09 11:31:06
+tags: 
+- Astro
+- JavaScript
+- frontend
+categories:
+- JavaScript
+- frontend
+- Astro
+twitter_text: "Creating A Pagination Component With Astro"
+authors: 
+- Ted Krueger
+image: https://images.unsplash.com/photo-1531177071211-ed1b7991958b?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NTd8fHNwYWNlJTIwc2hpcHxlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60
+image_url: https://unsplash.com/photos/b3PuuRU8IPc
+image_credit: Zoltan Tasi
+---
+
+Here at RIMdev, we're currently converting our static sites to use [Astro](https://astro.build/). As we switch from our current SSGs, [Hugo](https://gohugo.io/) and [Jekyll](https://jekyllrb.com/), we're creating reusable components that we'd like to document for the team. This series of posts will contain those components for not only our benefit but hopefully yours as well!
+
+Something that Hugo did well was Pagination. It was very simple to implement but it came with some limitations. Astro does not have a specific Pagination component but it _does_ give you the tools to make your own. Here's what I came up with and used on a couple of our sites for blog sections...
+
+## [TLDR;](#putting-it-together)
+<div class="mb-3 text--center">
+<img src="/images/get-to-the-point.gif" alt="Get to the point"/>
+</div>
+
+First, create your blog page within `/src/pages`. I like to add another folder here called `/blog` just to try to keep things a bit cleaner. So we have:
+
+```
+src
+- pages
+  - blog
+```
+
+Inside here add your page for the blog landing. This is the page that will show all of your posts, or at least what we want. You'll need the use brackets in this page title. Call it `[...page].astro`. We need the `[]` wrapping the name as this is a dynamic route.
+
+```
+src
+- pages
+  - blog
+    - [...page].astro
+```
+
+## [Blog Landing](#blog-landing)
+Now let's get to the guts of `[...page].astro`. You'll be leveraging `page.data` here. We also need 2 other components `PostSummary.astro`, which is a block of blog info like a title, image, and a teaser. You can add more to this if you need and I'll show you what this looks like in a bit. We also need a `Pagination.astro` component. This can also be customizzed to your needs but it will most likely always have the same data passed to it.
```

Comment:
```suggestion
Now let's get to the guts of `[...page].astro`. You'll be leveraging `page.data` here. We also need 2 other components:  `PostSummary.astro`, which is a block of blog info like a title, image, and a teaser. You can add more to this if you need, and I'll show you what this looks like in a bit. We also need a `Pagination.astro` component. This can also be customized to your needs, but it will most likely always have the same data passed to it.
```